ext/dom/element.c

```diff
@@ -728,57 +850,37 @@ PHP_METHOD(DOMElement, getAttributeNS)
 
 	DOM_GET_OBJ(elemp, id, xmlNodePtr, intern);
 
-	strattr = xmlGetNsProp(elemp, (xmlChar *) name, (xmlChar *) uri);
-
-	if (strattr != NULL) {
-		RETVAL_STRING((char *)strattr);
-		xmlFree(strattr);
+	bool should_free_result;
+	const xmlChar *result = dom_get_attribute_ns(intern, elemp, uri, uri_len, name, &should_free_result);
```

Comment:
I know that you are not accessing the `should_free_result`  unless `result != NULL` but that seems potentially error prone?

Reply:
I agree, let's initialize it to false.

Reply:
Ah seems this hasn't been done yet